### PR TITLE
Replace Travis with Github Actions

### PR DESCRIPTION
**Pull request by vogdb**
I see that js-libp2p-tcp also has [automerge.yml](https://github.com/libp2p/js-libp2p-tcp/blob/master/.github/workflows/automerge.yml). Should I add it here as well?

### CODE REVIEW

File: .github/workflows/js-test-and-release.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        node: [16]
+      fail-fast: true
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+      - uses: ipfs/aegir/actions/cache-node-modules@master
+      - run: npm install -g node-pre-gyp
```
Comment (vogdb) [inline .github/workflows/js-test-and-release.yml:37]: I've added installation of `node-pre-gyp` here. Let me know if I should move it elsewhere.

**Comment by vogdb** [comment]
Well, I've added `automerge.yml` in the end. Another question from me. `js-test-and-release.yml` is a typical github action that is used in the most of libp2p repos. It comes from https://github.com/protocol/.github/.  Now, it expects the project to have npm scripts: `test-node, test-chrome, test-chrome-webworker, test-firefox, test-firefox-webworker, test-electron-main, test-electron-renderer`. This project does not have them. Instead it has only `test-node, test-browser`. Should I keep only those two scripts in `js-test-and-release.yml`?

File: .github/workflows/js-test-and-release.yml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+          fetch-depth: 0
+      - uses: actions/setup-node@v2
+        with:
+          node-version: lts/*
+      - uses: ipfs/aegir/actions/cache-node-modules@master
+      - uses: ipfs/aegir/actions/docker-login@master
+        with:
+          docker-token: ${{ secrets.DOCKER_TOKEN }}
+          docker-username: ${{ secrets.DOCKER_USERNAME }}
+      - run: npm run --if-present release
```
Comment (wemeetagain) [inline .github/workflows/js-test-and-release.yml:165]: I think we would want the release process to be updated as well.
Take a look at the `package.json` from other libp2p repos, we'd want it to match.

**Review by wemeetagain — CHANGES_REQUESTED** [review]

File: .github/workflows/js-test-and-release.yml (hunk 2 above)
Comment (vogdb) [reply .github/workflows/js-test-and-release.yml:165]: If you mean integrating the semantic-release then I wanted to do it in a separate PR to keep things separate and clean. Currently working on migrating to Typescript. If you can, please answer my questions above.

**Comment by wemeetagain** [comment]
> Should I keep only those two scripts in js-test-and-release.yml?

I think you can leave the unused scripts since they're called with `--if-present` and seemingly won't break if they don't exist.

**Comment by vogdb** [comment]
Thank you for clarifying. I will add `test-browser` then as it's missing in CI but present in `package.json`

**Review by wemeetagain — APPROVED** [review]

**Comment by vogdb** [comment]
I can't merge so I will cc @vasco-santos

**Review by vasco-santos — APPROVED** [review]
LGTM! Thanks a lot @vogdb

File: .github/workflows/js-test-and-release.yml (hunk 1 above)
Comment (vasco-santos) [reply .github/workflows/js-test-and-release.yml:37]: @vogdb  It looks like this needs to run after `actions/setup-node@v2`

See https://github.com/libp2p/js-libp2p-webrtc-direct/runs/5537246479?check_suite_focus=true

Don't now why PR CI did not fail though

Comment (vogdb) [reply .github/workflows/js-test-and-release.yml:37]: Please look at #152. Did I understand you correctly?
Btw should be Github Actions enabled somehow to start working on PRs?

**Pull request merged**